lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -110,11 +109,33 @@ update the C<accounts> variable every 10 seconds
 async sub update_accounts {
     my $self = shift;
     while (1) {
+        await $self->get_hash_accounts();
         await $self->loop->delay_future(after => UPDATE_ACCOUNTS);
-        $self->{accounts} = await $self->rpc_client->accounts();
     }
 }
 
+=head2 get_hash_accounts
+
+Request the node accounts and convert it to a hash
+
+=over 4
+
+=back
+
+hash ref containing the accounts as keys
+
+=cut
+
+async sub get_hash_accounts {
+    my ($self) = @_;
+
+    my $accounts_response = await $self->rpc_client->accounts();
+    my %accounts = map { lc($_) => 1 } $accounts_response->@*;
+    $self->{accounts} = \%accounts;
+    undef $accounts_response;
```

Comment:
The variable drops out of scope after the next line, this really shouldn't be needed.